pkg/controller/controlplane/valuesprovider.go

```diff
@@ -365,16 +367,35 @@ func (vp *valuesProvider) GetControlPlaneShootChartValues(
 // GetStorageClassesChartValues returns the values for the storage classes chart applied by the generic actuator.
 func (vp *valuesProvider) GetStorageClassesChartValues(
 	_ context.Context,
-	_ *extensionsv1alpha1.ControlPlane,
+	cp *extensionsv1alpha1.ControlPlane,
 	cluster *extensionscontroller.Cluster,
 ) (map[string]interface{}, error) {
 	k8sVersionLessThan118, err := version.CompareVersions(cluster.Shoot.Spec.Kubernetes.Version, "<", "1.18")
 	if err != nil {
 		return nil, err
 	}
 
+	// Decode providerConfig
+	cpConfig := &apisaws.ControlPlaneConfig{}
+
+	if cp.Spec.ProviderConfig != nil {
+		if _, _, err := vp.Decoder().Decode(cp.Spec.ProviderConfig.Raw, nil, cpConfig); err != nil {
+			return nil, errors.Wrapf(err, "could not decode providerConfig of controlplane '%s'", util.ObjectName(cp))
+		}
+	}
+
+	encrypt := false
+
+	// User-provided encryption overrides the one set in the controller.
+	if cpConfig.ShootStorageClassConfig != nil && cpConfig.ShootStorageClassConfig.Encrypted != nil {
+		encrypt = *cpConfig.ShootStorageClassConfig.Encrypted
+	} else if vp.storageClass != nil && vp.storageClass.Encrypted != nil {
+		encrypt = *vp.storageClass.Encrypted
+	}
+
```

Comment:
I think somewhere here you would have to delete the existing `StorageClass` as it's immutable.

Reply:
Ref https://github.com/gardener/gardener-extension-provider-openstack/pull/102